src/brsTypes/components/RoSGNode.ts

```diff
@@ -296,7 +296,7 @@ function isSubtypeCheck(currentNodeType: string, checkType: string): boolean {
 export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
     readonly kind = ValueKind.Object;
     private fields = new Map<string, Field>();
-    private children: RoSGNode[] = [];
+    public children: RoSGNode[] = [];
```

Comment:
I wouldn't want _children_ directly exposed as public. This could lead to bug down the line where the children array is altered, but we don't, for example, fire the callbacks for listeners watching the children. 

Instead I'd create a utility function like this:
```
private function sliceChildren(numChildren, index) : RoSGNode[] {
            // stolen from getChildren()
            let numChildrenValue = num_children.getValue();
            let indexValue = index.getValue();
            let childrenSize = this.children.length;
            if (numChildrenValue <= -1 && indexValue === 0) {
                //short hand to return all children
                return this.children.slice();
            } else if (numChildrenValue <= 0 || indexValue < 0 || indexValue >= childrenSize) {
                //these never return any children
                return [];
            } else {
                //only valid cases
                return this.children.slice(indexValue, indexValue + numChildrenValue);
            }

}
```

And then the `getChildren` callable would just look like this:
```
return RoArray(this.sliceChildren(...));
```

Reply:
@strattonbrazil Thanks, I was not sure if I was on the right track there so that's helpful! 
So, exposing `children` the way I did for RoSGNode would be more dangerous than how [`children` are exposed](https://github.com/sjbarag/brs/blob/0505973a011453e41bca4bc538ad1364b73413fc/src/componentprocessor/index.ts#L56) in the [AddChildren](https://github.com/sjbarag/brs/blob/0505973a011453e41bca4bc538ad1364b73413fc/src/brsTypes/components/RoSGNode.ts#L1897) method above for the `ComponentNode`? I am wondering how the two situations are different for my understanding.

Reply:
Sure, as I understand it the input to _addChildren_ references an object that has a children array and is basically just a block of data transporting between the interpreter and this function. It can be used and modified as its about to be descoped. 

In the case of _RoSGNode_, that _children_ reference is actually a member variable and by restricting access to it we can provide more safeguards to how its accessed. In the example I gave not letting an outside component insert, remove, or clear children without doing any necessary auxiliary tasks. 

For example:
```
    private children = ...
    public function appendChild(child) {
        children.append(child);

        // bookkeeping like to notify observers of this node's _change_ interface that its children have been altered
        ...
    }
```

If one were to expose children directly it introduces opportunities for parts of the code outside RoSGNode to avoid any necessary code for maintaining and messaging children.

Reply:
Ah I see. Thank you!

Reply:
I decided not to pursue this refactor due to time constraints :)